Blizzard.IO.RabbitMQ.Builders/NetqRabbitConsumerBuilder.cs

```diff
@@ -0,0 +1,107 @@
+        private IAbstractTypeDeserializer<TData> _abstractDeserializer = new JsonAbstractTypeDeserializer<TData>();
+        private IConverter<MessageProperties, RabbitMessageProperties> _converter;
+        private bool _isAbstract = false;
+
+        public NetqRabbitConsumerBuilder(ILoggerFactory loggerFactory) 
+            : base(loggerFactory)
+        {
+        }
+
+        public NetqRabbitConsumerBuilder<TData> AddRabbitConnectionProperties(
+            ushort timeout = 10,
+            string product = null,
+            string platform = null,
+            string virtualHost = null,
+            int requestHeartbeat = 10,
+            int prefetchCount = 50,
+            bool publisherConfirms = false,
+            bool persistentMessages = true)
+        {
+            Timeout = timeout;
+            PrefetchCount = prefetchCount;
+            Product = product;
+            Platform = platform;
+            PublisherConfirms = publisherConfirms;
+            RequestHeartbeat = requestHeartbeat;
+            VirtualHost = virtualHost;
+            PersistentMessages = persistentMessages;
+            return this;
+        }
+
+        public NetqRabbitConsumerBuilder<TData> AddHostnameAndCredentials(string hostname, string username, string password)
+        {
+            Hostname = hostname;
+            Username = username;
+            Password = password;
+            return this;
+        }
+
+        public NetqRabbitConsumerBuilder<TData> AddCredentials(string username, string password)
+        {
+            Username = username;
+            Password = password;
+            return this;
+        }
+
+        public NetqRabbitConsumerBuilder<TData> AddDeserializer(IDeserializer<TData> deserializer)
+        {
+            _deserializer = deserializer;
+            return this;
+        }
+
+        public NetqRabbitConsumerBuilder<TData> AddAbstractDeserializer(IAbstractTypeDeserializer<TData> abstractDeserializer)
+        {
+            _abstractDeserializer = abstractDeserializer;
+            return this;
+        }
+
+        public NetqRabbitConsumerBuilder<TData> AddAbstractSupport()
+        {
+            _isAbstract = true;
+            return this;
+        }
+
+        public NetqRabbitConsumerBuilder<TData> RemoveAbstractSupport()
+        {
+            _isAbstract = false;
+            return this;
+        }
+
+        public NetqRabbitConsumerBuilder<TData> AddSourceQueue(RabbitQueue sourceQueue)
+        {
+            _sourceQueue = sourceQueue;
+            return this;
+        }
+
+        public NetqRabbitConsumerBuilder<TData> AddConverter(IConverter<MessageProperties, RabbitMessageProperties> converter)
+        {
+            _converter = converter;
+            return this;
+        }
+
+        public NetqRabbitConsumer<TData> Build()
+        {
+            IBus bus = InitConnection();
+            _sourceQueue = _sourceQueue ?? bus.DeclareQueue("DefaultQueue");
```

Comment:
Create another variable dont use the field again it is less readable like that

Reply:
`??` is a c# feature, why not to use it?

Reply:
use this feature but assign the result to a new varaible it is confusing and looks weird to assign the same variable to itself

Reply:
f